### ISSUE TITLE
symbol-overlay like persistent highlights

### ISSUE THREAD

**Issue by mpenet** [emacs-lsp/lsp-mode#3103]
[symbol-overlay ](https://github.com/wolray/symbol-overlay) has the ability to highlight symbols at point, a bit like lsp does; lsp does it better given it knows more about the symbol.
That part is covered, but symbol-overlay also adds the ability to make some highlights persistent after some keypress, with a different color per highlight (same symbols would use the same color). This is **very** useful for code reviews/reading. I supose since the method for highlighting is similar between the 2 codebases (overlays) it would be possible to integrate this to lsp-mode as well. 

Right now I rely on both modes to get what I want, I disabled idle highlights from symbol overlay in favor of lsp-symbol-highlight and use symbol overlay for the persistent feature, but that is brittle.

**Comment by ericdallo** [comment]
@yyoncho do you think that would be possible to make an integration just like we did with iedit?

**Comment by yyoncho** [comment]
> @yyoncho do you think that would be possible to make an integration just like we did with iedit?

It seems like it will be possible with small adjustments on symbol-overlay side to allow plugging lsp-mode (cc @purcell)

**Comment by purcell** [comment]
I've actually been working on a pluggable highlighting library recently that might fit the bill for this...

**Comment by purcell** [comment]
Yes, I worked a little more on it recently, hoping to make an initial release soon!
